aicsimageio/readers/glob_reader.py

```diff
@@ -0,0 +1,541 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from typing import Any, Dict, List, Optional, Tuple, Union, Callable
+
+import re
+import glob
+from collections import OrderedDict
+from pathlib import Path
+import dask.array as da
+import numpy as np
+import pandas as pd
+import xarray as xr
+from dask import delayed
+from fsspec.spec import AbstractFileSystem
+from tifffile import TiffFile, TiffFileError, imread
+from tifffile.tifffile import TiffTags
+
+from .. import constants, exceptions, types
+from ..dimensions import (
+    DEFAULT_CHUNK_DIMS,
+    REQUIRED_CHUNK_DIMS,
+    DimensionNames,
+    DEFAULT_DIMENSION_ORDER,
+    DEFAULT_DIMENSION_ORDER_LIST_WITH_SAMPLES,
+)
+from ..metadata import utils as metadata_utils
+from ..utils import io_utils
+from .reader import Reader
+
+
+TIFF_IMAGE_DESCRIPTION_TAG_INDEX = 270
+
+
+class GlobReader(Reader):
+
+    """
+    Wraps the tifffile imread API to provide the same aicsimageio Reader API but for
+    multifile tiff datasets (and other tifffile supported) images.
+
+    Parameters
+    ----------
+    glob_in: Union[str, List[str]]
+        Glob string that identifies all files to be loaded or a list
+        of paths to the files as returned by glob.
+    indexer: Union[Callable, pandas.DataFrame]
+        If callable, should consume each filename and return a pd.Series with series index
+        corresponding to the dimensions and values corresponding to the array index of that image file within the larger array.
+        Default: None (Look for 4 numbers in the file name and use them as S, T, C, and Z indices.
+    scene_glob_character: str
```

Comment:
why is there a scene glob character but not a time or channel glob character?

Reply:
This is a good question but personally has anyone ever called channels anything besides `C`?

The reason for scene glob char was because I originally commented and said "for code clarity can we not use `DimensionNames.Samples` for the "Scene character" parsing / splitting and instead just parameterize it so that someone can use maybe `Image_` or `I_` or something besides "S""?